packages/query-core/src/types.ts

```diff
@@ -36,6 +36,16 @@ export type Override<TTargetA, TTargetB> = {
 
 export type NoInfer<T> = [T][T extends any ? 0 : never]
 
+export type DropLast<T extends ReadonlyArray<unknown>> = T extends readonly [
+  ...infer R,
+  unknown,
+]
+  ? R
+  : never
+
+export type TuplePrefixes<T extends ReadonlyArray<unknown>> =
```

Comment:
all those types become part of the public interface, I don’t think we want that so please just inline them in `utils.ts`

Reply:
That's a good thought! I didn't even think of that. Updated.